strongbox-cron-tasks/src/main/java/org.carlspring.strongbox/cron/api/jobs/RebuildMetadataCronJob.java

```diff
@@ -0,0 +1,171 @@
+                List<String> repositoriesId = getRepositoriesId(storagePath);
+                rebuildRepositories(repositoriesId, storageId, basePath);
+            }
+            else
+            {
+                artifactMetadataService.rebuildMetadata(storageId, repositoryId, basePath);
+            }
+        }
+        catch (IOException | XmlPullParserException | NoSuchAlgorithmException e)
+        {
+            e.printStackTrace();
+            logger.error("IOException: ", e);
+        }
+        manager.addExecutedJob(config.getName(), true);
+    }
+
+    /**
+     * To rebuild artifact's metadata in repositories
+     *
+     * @param repositories list of repositories
+     * @param storageId    path of storage
+     * @param basePath     basePath of artifact
+     * @throws NoSuchAlgorithmException
+     * @throws XmlPullParserException
+     * @throws IOException
+     */
+    private void rebuildRepositories(List<String> repositories,
+                                     String storageId,
+                                     String basePath)
+            throws NoSuchAlgorithmException, XmlPullParserException, IOException
+    {
+        for (String repository : repositories)
+        {
+            artifactMetadataService.rebuildMetadata(storageId, repository, basePath);
+        }
+    }
+
+    /**
+     * To get list of repositoryId in storage
+     *
+     * @param storage
+     * @return list of repositoryId
+     */
+    private List<String> getRepositoriesId(Path storage)
+    {
+        File file = storage.toFile();
+        return getListDirectories(file);
+    }
+
+    /**
+     * To get list of storageId
+     *
+     * @return list of storageId
+     */
+    private List<String> getStoragesId()
+    {
+        File file = Paths.get(storagesBasePath).toFile();
+        return getListDirectories(file);
+    }
+
+    /**
+     * To get list of directories
+     *
+     * @param baseFile Object File
+     * @return list of directories
+     */
+    private List<String> getListDirectories(File baseFile)
+    {
+        List<String> listDirectories = new ArrayList<>();
+        File[] listFiles = baseFile.listFiles();
+
+        if (listFiles != null)
+        {
+            for (File file : listFiles)
+            {
+                if (file.isDirectory())
+                {
+                    String directoryId = file.getAbsolutePath().substring(baseFile.getAbsolutePath().length() + 1);
+                    listDirectories.add(directoryId);
+                }
+            }
+        }
+        return listDirectories;
+    }
+
+    /**
+     * To get base path of storages
+     *
+     * @return String base path
+     */
+    private static String getStoragesBasePath()
+    {
+        return Paths.get(ConfigurationResourceResolver.getVaultDirectory(), BASEDIR_STORAGES).toString();
+    }
+
+    private ArtifactMetadataServiceImpl getArtifactMetadataService()
```

Comment:
@kate-novik , @AlexOreshkevich :

Please, remove this method. With proper autowiring, this will not be required.